src/app/pages/home/create-contact-dialog/create-contact-dialog.component.ts

```diff
@@ -43,16 +63,20 @@ export class CreateContactDialogComponent {
 
   async save() {
     const data = this.contactForm.getRawValue();
-      try {
-        await this.contactProvider.saveNewContact(data);
-      } catch (error: any) {
-        console.log('ERROR 132' + error);
+    try {
+      await this.contactProvider.saveNewContact(data);
+      this.onChange.emit();
+      this.snackbarService.successMessage("Contato criado com sucesso");
+      this.dialogRef.close();
+    } catch (error: any) {
+      if (error?.response?.data?.errors) {
+        this.errors = error.response.data.errors;
       }
```

Comment:
o que acontece nesse trecho de código caso ocorra algum erro durante a requisição de criação de novo contato? se construímos bem o backend, a api vai retornar um objeto '{ apiResponse: {}, errors: ['mensagem de erro'] } o ideial aqui seria vc atribuir o resultado da chamada da linha 67 à uma variável e antes de tomar qlquer ação verificar se existem erros na propriedade 'errors' do objeto retornado, caso não haja erros (se errors.length > 0 for false) dai sim tomar as ações das linhas 68-70, e no bloco catch deve ficar apenas um tratamento de erro genérico, para o caso do backend estar indisponivel ou outra questão que faça o código quebrar, dai dentro do bloco catch poderia ser feito algo como

catch {
      this.errors = { message: "Ocorreu um erro ao criar seu contato. Tente novamente." }; //Verificar se a sintaxe está correta
      }

fazer isso para todos os componentes que fazem chamadas de endpoint!